packages/enhanced-img/src/preprocessor.js

```diff
@@ -0,0 +1,188 @@
+import MagicString from 'magic-string';
+import { parse } from 'svelte-parse-markup';
+import { walk } from 'svelte/compiler';
+
+const ASSET_PREFIX = '___ASSET___';
+
+// TODO: expose this in vite-imagetools rather than duplicating it
+const OPTIMIZABLE = /^[^?]+\.(avif|heif|gif|jpeg|jpg|png|tiff|webp)(\?.*)?$/;
+
+/**
+ * @returns {import('svelte/types/compiler/preprocess').PreprocessorGroup}
+ */
+export function image() {
+	return {
+		markup({ content, filename }) {
+			const s = new MagicString(content);
+			const ast = parse(content, { filename });
+
+			// Import path to import name
+			// e.g. ./foo.png => ___ASSET___0
+			/** @type {Map<string, string>} */
+			const imports = new Map();
+
+			/**
+			 * @param {import('svelte/types/compiler/interfaces').TemplateNode} node
+			 * @param {{ type: string, start: number, end: number, raw: string }} src_attribute
+			 */
+			function update_element(node, src_attribute) {
+				if (src_attribute.type === 'MustacheTag') {
```

Comment:
just flagging that the shape of the AST will change in Svelte 5 (but behind an option until Svelte 6) — `MustacheTag` will become `ExpressionTag`. nothing to change for now though